functions/Test-DbaSqlLoginWeakPassword.ps1

```diff
@@ -0,0 +1,131 @@
+
+            Test all SQL logins that the password is null, same as username and defined in the csv file on SQL server instance Dev01
+
+    #>
+    [CmdletBinding()]
+    Param (
+        [parameter(Position = 0, Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer", "SqlServers")]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [String[]]$SQLPass,
+        [String]$Path,
+        [Alias('Silent')]
+        [switch]$EnableException
+    )
+
+    begin {
+        Test-DbaDeprecation -DeprecatedOn 1.0.0 -Parameter Detailed
+
+        $CheckPasses = "''", "'@@Name'"
+
+        if ($SQLPass) {
+            $SQLPass | ForEach-Object {$CheckPasses += "'" + $psitem + "'" }
+        }
+        if ($Path) {
+            Import-Csv -Path $Path -Header 'Password' | ForEach-Object { $CheckPasses += "'" + $psitem.password + "'" }
+        }
+
+        foreach ($CheckPass in $CheckPasses) {
+            if ($CheckPasses.IndexOf($CheckPass) -eq 0) {
+                $checks = "SELECT " + $CheckPass
+            } else {
+                $checks += "
+        UNION SELECT " + $CheckPass
+            }
+        }
+
+        $sql = "DECLARE @WeakPwdList TABLE(WeakPwd NVARCHAR(255))
+            --Define weak password list
+            --Use @@Name if users password contain their name
+            INSERT INTO @WeakPwdList(WeakPwd)
+            $checks
+
+            SELECT @@SERVERNAME As ServerName,
+                SysLogins.name as SqlLogin,
+                REPLACE(WeakPassword.WeakPwd,'@@Name',SysLogins.name) As [Password],
+                SysLogins.is_disabled as Disabled,
+                SysLogins.create_date as CreatedDate,
+                SysLogins.modify_date as ModifiedDate,
+                SysLogins.default_database_name as DefaultDatabase
+            FROM sys.sql_logins SysLogins
+            INNER JOIN @WeakPwdList WeakPassword ON (PWDCOMPARE(WeakPassword.WeakPwd, password_hash) = 1
+                OR PWDCOMPARE(REPLACE(WeakPassword.WeakPwd,'@@Name',SysLogins.name),password_hash) = 1)"
+
+
+    }
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential
+                Write-Message -Message "Connected to: $instance." -Level Verbose
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+            if ($Server.VersionMajor -lt 10) {
+                throw "PWDCOMPARE is only supported on sql 2008 and up. Quitting."
+            }
+            Write-Message -Level Debug -Message "Executing $sql"
+            Write-Message -Level Verbose -Message "Testing: same username as Password"
+            Write-Message -Level Verbose -Message "Testing: the following Passwords $CheckPasses"
+            $server.Query("$sql")
```

Comment:
The output should include ComputerName, InstanceName, and SqlInstance (see other commands line Get-DbaLogin).

Reply:
Added ComputerName, InstanceName and SqlInstance. Did it from the tsql since I can get multiple results on one server. 

I do not know if that makes any difference? I don't call Select-DefaultView either do I need to?